From 47a17788ac3c7dd17017592558cba626ebf0234e Mon Sep 17 00:00:00 2001
From: Nael Masalha <nael.masalha@intel.com>
Date: Sun, 24 Oct 2021 11:42:50 +0300
Subject: [PATCH] Enable Y8 on separate video node

 - This solution is based on HW short of dedicated I2C bus

Signed-off-by: Nael Masalha <nael.masalha@intel.com>
---
 common/tegra194-camera-d4xx.dtsi | 97 ++++++++++++++++++++------------
 1 file changed, 62 insertions(+), 35 deletions(-)

diff --git a/common/tegra194-camera-d4xx.dtsi b/common/tegra194-camera-d4xx.dtsi
index f85fc1d..c394d6f 100644
--- a/common/tegra194-camera-d4xx.dtsi
+++ b/common/tegra194-camera-d4xx.dtsi
@@ -46,14 +46,6 @@
 									data-types = <0x31>;
 									vc-id = <0>;
 								};
-								src_y0: source@1 {
-									data-types = < 0x2a
-											0x1e
-											0x2b
-											0x32
-											0x24>;
-									vc-id = <0>;
-								};
 							};
 						};
 					};
@@ -80,27 +72,6 @@
 						default_framerate = "5";
 						vc_id = "0";
 					};
-					mode1 {
-						pixel_t = "grey_y8";
-						num_lanes = "2";
-						active_w = "1280";
-						active_h = "720";
-						tegra_sinterface = "serial_e";
-						mclk_khz = "24000";
-						/*pix_clk_hz = "148500000";*/ /* for 2 streams? */
-						/*pix_clk_hz = "74250000";*/
-						/*pix_clk_hz = "375000000";*/
-						pix_clk_hz = "74250000";
-						discontinuous_clk = "no";
-						dpcm_enable = "false";
-						cil_settletime = "0";
-						line_length = "1280"; /* 2200 */
-						mclk_multiplier = "15.625";
-						embedded_metadata_height = "1";
-						min_framerate = "5";
-						max_framerate = "30";
-						default_framerate = "5";
-					};
 				};
 			};
 			i2c@1 { //RGB0
@@ -156,6 +127,61 @@
 					};
 				};
 			};
+			i2c@2 {
+				reg = <0x4>;	// line 4 of the I2C switch
+				#address-cells = <0x1>;
+				i2c-mux,deselect-on-exit;
+				#size-cells = <0x0>;
+
+				d4m2: d4m@10 {
+					status = "ok";
+					reg = <0x10>;
+					compatible = "intel,d4xx";
+					vcc-supply = <&p2822_vdd_1v8_cvb>;
+					/*reset-gpios = <&tegra_main_gpio CAM0_RST_L GPIO_ACTIVE_HIG>;*/
+					cam-type = "Y8";
+					ports {
+						#address-cells = <1>;
+						#size-cells = <0>;
+
+						port@0 {
+							reg = <0>;
+							d4m2_out: endpoint {
+								port-index = <1>;
+								bus-width = <2>;
+								remote-endpoint = <&csi_in2>;
+								src_y80: source@0 {
+									data-types = <0x2a>;
+									vc-id = <2>;
+								};
+							};
+						};
+					};
+					/* mode0: Y8, mode1: depth D16 */
+					mode0 {
+						pixel_t = "grey_y8";
+						num_lanes = "2";
+						active_w = "1280";
+						active_h = "720";
+						tegra_sinterface = "serial_b";
+						mclk_khz = "24000";
+						/*pix_clk_hz = "148500000";*/ /* for 2 streams? */
+						/*pix_clk_hz = "37125000";*/
+						/*pix_clk_hz = "375000000";*/
+						pix_clk_hz = "74250000";
+						discontinuous_clk = "no";
+						dpcm_enable = "false";
+						cil_settletime = "0";
+						line_length = "1280"; /* 2200 */
+						mclk_multiplier = "15.625";
+						embedded_metadata_height = "1";
+						min_framerate = "5";
+						max_framerate = "30";
+						default_framerate = "30";
+						vc_id = "2";
+					};
+				};
+			};
 
 		};
 	};
@@ -229,7 +255,7 @@
 			};
 			channel@2 {
 				reg = <0x2>;
-				status = "disabled";
+				status = "ok";
 
 				ports {
 					#address-cells = <0x1>;
@@ -241,9 +267,9 @@
 
 						csi_in2: endpoint@4 {
 							status = "ok";
-							port-index = <2>;
+							port-index = <0>;
 							bus-width = <2>;
-							remote-endpoint = <&d4m1_out/*&d4m2_out0*/>;
+							remote-endpoint = <&d4m2_out>;
 						};
 					};
 					port@1 {
@@ -378,12 +404,13 @@
 				};
 				port@2 {
 					reg = <2>;
-					status = "disabled";
+					status = "ok";
 
 					vi_in2: endpoint {
-						status = "disabled";
+						status = "ok";
 						port-index = <0>;
 						bus-width = <2>;
+						vc-id = <2>;
 						remote-endpoint = <&csi_out2>;
 					};
 				};
@@ -427,7 +454,7 @@
 					remote-source = <&src_depth0>;
 				};
 				source@1 {
-					remote-source = <&src_y0>;
+					remote-source = <&src_y80>;
 				};
 				//source@2 {
 				//	remote-source = <&src_depth1>;
-- 
2.17.1

